### PR TITLE
MAGN-7776 No message is given by the ui when creating a Preset with an already assigned name

### PR DESCRIPTION
**Pull request by ramramps**
### Purpose
 This PR fixes the issue with same preset names

YT: http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7776

### Declarations
- [x] The code base is in a better state after this PR
Added a new "overwrite" dialog window that ensures Preset names are unique.
          
- [] The level of testing this PR includes is appropriate
No tests were added. The changes are made on the View.
           

### Reviewers
- [ ] @pboyer

### CODE REVIEW

File: src/DynamoCore/Models/WorkspaceModel.cs [hunk 1]
```diff
@@ -233,7 +233,7 @@ protected virtual void OnConnectorDeleted(ConnectorModel obj)
         /// <summary>
         ///     A set of input parameter states, this can be used to set the graph to a serialized state.
         /// </summary>
-        public IEnumerable<PresetModel> Presets { get { return presets;} }
+        public List<PresetModel> Presets { get { return presets;} }
```
Comment (pboyer) [inline src/DynamoCore/Models/WorkspaceModel.cs]: This change doesn't work.  :( A  List is a mutable type (i.e. you can call Add on it).  IEnumerable is not.  This type was chosen specifically for this reason.

File: src/DynamoCoreWpf/UI/Prompts/PresetPrompt.xaml.cs [hunk 2]
```diff
@@ -24,6 +23,36 @@ public PresetPrompt()
 
         void OK_Click(object sender, RoutedEventArgs e)
         {
+            if (this.Owner != null)
+            {
+                var dataContext = this.Owner.DataContext as DynamoViewModel;
+                 //get the preset names from workspace
+                if (dataContext != null && dataContext.Model.CurrentWorkspace.Presets.Any())
```
Comment (pboyer) [inline src/DynamoCoreWpf/UI/Prompts/PresetPrompt.xaml.cs]: Why don't we store `dataContext.Model.CurrentWorkspace` in a temporary variable?

File: src/DynamoCoreWpf/UI/Prompts/PresetPrompt.xaml.cs [hunk 3]
```diff
@@ -24,6 +23,36 @@ public PresetPrompt()
...
+                        if (newDialog.ShowDialog() != true)
+                        {                                                        
+                            e.Handled = true;
+                            return;
+                        }
+                        //If the dialog result is true, then remove the old preset
+                        else
+                        {
+                           var oldPreset = dataContext.Model.CurrentWorkspace.Presets.FirstOrDefault(x => x.Name == Text);
+                           dataContext.Model.CurrentWorkspace.Presets.Remove(oldPreset);
```
Comment (pboyer) [inline src/DynamoCoreWpf/UI/Prompts/PresetPrompt.xaml.cs]: Please rename `WorkspaceModel.RemoveState` to `WorkspaceModel.RemovePreset` and use that API instead.

**Comment by ramramps** [comment]
Ping @pboyer

**Pull request merged**